src/app/routes/dashboard/dashboard.component.ts

```diff
@@ -62,6 +64,12 @@ export class DashboardComponent implements AfterViewInit {
   }
 
   ngAfterViewInit(): void {
+    this.activeRoute.queryParams.subscribe(async (params: any) => {
```

Comment:
@dawidgil Is this wrong how I implemented this URL param interception? Felt like if I went right to auth guard and added the /vp route, it always re-routed back to dashboard, and the return URL was not correct. LMK how you would handle this better!

Reply:
The problem here is with race condition. You have 2 things that are going on at once:
1. Checking url param and navigating - which is fast.
2. Reinitialization authentication - which is slow because it has to call backend endpoints.
The best would be to not care about authentication here. Just navigate to the `vp` route and let that page handle everything -> like in enrol page. When you are not logged in, then it displays dialog with options to login. If you're logged in then open `vp` page or try to reinitialize auth when state is not stored in app. Example how it looks like: `https://switchboard-dev.energyweb.org/enrol?app=test.apps.dawidgil.iam.ewc&roleName=user`. For now you can just copy paste it, as this is very big component and i didn't refactored it fully yet :/